src/main/java/redis/clients/jedis/JedisSentinelPool.java

```diff
@@ -175,6 +175,27 @@ public JedisSentinelPool(String masterName, Set<String> sentinels,
     initPool(master);
   }
 
+  public JedisSentinelPool(String masterName, Set<String> sentinels, final GenericObjectPoolConfig poolConfig, final JedisFactory factory) {
+    this.poolConfig = poolConfig;
+    this.connectionTimeout = Protocol.DEFAULT_TIMEOUT;
+    this.soTimeout = Protocol.DEFAULT_TIMEOUT;
+    this.infiniteSoTimeout = 0;
+    this.user = null;
+    this.password = null;
+    this.database = Protocol.DEFAULT_DATABASE;
+    this.clientName = null;
+    this.sentinelConnectionTimeout = Protocol.DEFAULT_TIMEOUT;
+    this.sentinelSoTimeout = Protocol.DEFAULT_TIMEOUT;
+    this.sentinelUser = null;
+    this.sentinelPassword = null;
+    this.sentinelClientName = null;
+    this.factory = factory;
+
+    HostAndPort master = initSentinels(sentinels, masterName);
+    initPool(poolConfig, factory);
```

Comment:
@sazzad16 perhaps I get it wrong, but do you need to call initPool twice?

Reply:
@gkorland `initPool(poolConfig, factory);` was usually get called within `initPool(master);` (Line 219). But as we are doing `this.factory = factory;` (Line 192) now, the `if (factory == null)` check (Line 216) would not let `initPool(poolConfig, factory);` to happen at usual place. So we're doing it here.

Reply:
Please at least leave a comment in the code explaining this double call.